File: AI.Core/OperatorsImplimentation/BasicOperators/Equal.cs
namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    class Equal : Operator
    {
        /// <summary>
        /// Just repeats Enter on exit
        /// </summary>
        public Equal(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 1;
        }

        public override void Action()
        {
            ExitContacts.SetValue(EnterContacts[0].Value);
        }
    }
}

File: NativeCore/Equal.cpp
#include "Equal.h"
#include "stdafx.h"

namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Equal::Equal(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(1);
				}

				void Equal::Action()
				{
					ExitContacts->SetValue(EnterContacts[0]->Value);
				}
			}
		}
	}
}
